### PR TITLE
add python 3.8.0rc1, refactor

### PR DESCRIPTION
**Pull request by mattip**
Update the Dockerfile so that hopefully future python versions will not require a complete rebuild (i.e. the move from python `3.8.0rc1` to `3.8.0`). Also add 3.8.0rc1 to the image

### CODE REVIEW

File: choose_python.sh [hunk 1]
```diff
@@ -3,14 +3,23 @@
...
+    if [ ${py_nodot} -ge "38" ]; then
+        abi_suff=""
+    fi
+    py_bin=/oopt/cp${py_nodot}${abi_suff}/bin/python$py_ver
 else
     py_bin=/usr/bin/python$py_ver
 fi
+if [ ! -e py_bin ]; then
+    exit 1
+fi
```
Comment (mattip) [inline choose_python.sh:22]: i am not sure what happens if this script exits, since it is a wrapper

**Comment by mattip** [comment]
I verified that this image works with the numpy-wheels build locally. Are there any further checks I can do for validity?

**Comment by charris** [comment]
Needs rebase.

**Comment by matthew-brett** [comment]
Sorry - yes - I merged this, with another commit on top - thanks.

I think the merge generated the 64-bit container, and I pushed up the 32-bit container manually - so the containers should be good to go now.

**Comment by charris** [comment]
Still not working
```
Status: Downloaded newer image for matthewbrett/trusty:32
The path /usr/bin/python3.8 (from --python=/usr/bin/python3.8) does not exist
```
Is it getting the image from the right place? How to check if the images are correctly updated?

**Comment by matthew-brett** [comment]
It's updated, but I forgot to update the script that activates Python, to take into account versions greater than 3.7.   Should be done in 20 minutes or so.

**Comment by matthew-brett** [comment]
Oops - no - sorry - I need to fix my fix.  Will post here when done / uploaded.

**Comment by mattip** [comment]
Somehow 6dc8eaf didn't make it into the merge

**Comment by matthew-brett** [comment]
Yes, sorry, I'm not sure what happened there.   I think I've caught up with you now though.

**Comment by matthew-brett** [comment]
I did manual uploads.  Travis build should find them now.

**Comment by charris** [comment]
Yay, it worked, both 64 and 32 bits. I'll make a push to numpy-wheels master so we can get some testing. What do we need for Mac and Windows?

**Pull request closed without merging**